DeepSpeech.py

```diff
@@ -446,6 +446,10 @@ def train():
 
     # Building the graph
     optimizer = create_optimizer()
+    
+    # Enable mixed precision training
+    if FLAGS.automatic_mixed_precision:
+        optimizer = tf.train.experimental.enable_mixed_precision_graph_rewrite(optimizer)
```

Comment:
@vinhngx  This is under experimental, could you please elaborate / link to further docs on what one can expect / not expect from this ?

Reply:
Hi @lissyx, though under experimental, automatic mixed precision has been tested quite extensively and incorporated into the TF official model zoo (https://github.com/tensorflow/models/pull/7419, https://github.com/tensorflow/models/pull/7531), tensor2tensor (https://github.com/tensorflow/tensor2tensor/pull/1680), NVIDIA TF model zoo (https://github.com/NVIDIA/DeepLearningExamples/tree/master/TensorFlow). Most of the time, AMP provides speed up at little or no loss to model accuracy. 

The docstring of this function is a good place of reference:
https://github.com/tensorflow/tensorflow/pull/29249/files